### PR TITLE
Fix test tagging and add interfaces to documentation

### LINKED ISSUE #197: Tests that are not mapped to API methods
We have a few tests which are not mapped to the API:

```
$ npm run docs
...
unable to find the method ssf.MessageService.receive within the typescript documentation
unable to find the method ssf.Window() within the typescript documentation
unable to find the method ssf.Window(center) within the typescript documentation
unable to find the method ssf.Window(maxHeight) within the typescript documentation
unable to find the method ssf.Window(minHeight) within the typescript documentation
unable to find the method ssf.Window(x) within the typescript documentation
unable to find the method ssf.Window(y) within the typescript documentation
unable to find the method ssf.Window.isVisible within the typescript documentation
unable to find the method ssf.Window.maximized within the typescript documentation
unable to find the method ssf.WindowEvent.blur within the typescript documentation
unable to find the method ssf.WindowEvent.closed within the typescript documentation
unable to find the method ssf.WindowEvent.focus within the typescript documentation
unable to find the method ssf.WindowEvent.hide within the typescript documentation
unable to find the method ssf.WindowEvent.maximize within the typescript documentation
unable to find the method ssf.WindowEvent.minimize within the typescript documentation
unable to find the method ssf.WindowEvent.move within the typescript documentation
unable to find the method ssf.WindowEvent.resize within the typescript documentation
unable to find the method ssf.WindowEvent.restore within the typescript documentation
unable to find the method ssf.WindowEvent.show within the typescript documentation
```

### PR DESCRIPTION
**Pull request by BenLambertNcl**
Add better tags for tests to partially fix #197, and add interfaces into documentation. Example:
image

Adding the test data to the docs still needs to be added, but this fixes the errors for the test data not being found.

### CODE REVIEW

File: packages/api-specification/transform-type-info.js [hunk 1]
```diff
@@ -63,12 +63,25 @@ if (program.testfile) {
     Object.keys(testReport).forEach((testMethod) => {
       const results = testReport[testMethod];
       const [, className, methodName] = testMethod.split('.');
-      const method = jspath.apply(`..{.kindString === "Class" && .name === "${className}"}.children{.kindString === "Method" && .name === "${methodName}"}`, typeInfo);
+      const method = jspath.apply(`..{(.kindString === "Class" || .kindString === "Interface") && .name === "${className}"}.children{.kindString === "Method" && .name === "${methodName}"}`, typeInfo);
       if (method.length > 0) {
         method[0].signatures[0].results = results;
-      } else {
-        console.warn(`unable to find the method ${testMethod} within the typescript documentation`);
```
Comment (JohnEz) [inline packages/api-specification/transform-type-info.js]: Why did we want to remove this else?

File: packages/api-specification/transform-type-info.js [hunk 2]
```diff
@@ -63,12 +63,25 @@ if (program.testfile) {
...
+        return;
       }
+      const property = jspath.apply(`..{(.kindString === "Class" || .kindString === "Interface") && .name === "${className}"}.children{.kindString === "Property" && .name === "${methodName}"}`, typeInfo);
```
Comment (AndyMJones) [inline packages/api-specification/transform-type-info.js]: There is some repetition here, might be worth pulling it out.

File: packages/api-specification/transform-type-info.js [hunk 3]
```diff
@@ -119,6 +138,17 @@ const documentClass = (className) => {
           jspath.apply('.{.kindString === "Method"}', d.match.children).length && section(d.runner(jspath.apply('.{.kindString === "Method"}', d.match.children)), {class: 'methods'})], {id: className, class: 'docs-title'})
     ),
     jsont.pathRule(
+      '.{.kindString === "Interface"}', d =>
+        section([
+          h2(d.match.name),
+          p(formatComment(d.match.comment)),
+          // create the various sections by filtering the children based on their 'kind'
+          jspath.apply('.{.kindString === "Property"}', d.match.children).length && h3('Properties'),
```
Comment (AndyMJones) [inline packages/api-specification/transform-type-info.js]: More repetition here

**Review by ColinEberhardt — APPROVED** [review]
Thanks @BenLambertNcl looks good to me.

When merged, can you update https://github.com/symphonyoss/ContainerJS/issues/197 with the remaining tests that are not associated with API methods? thanks

**Comment by BenLambertNcl** [comment]
Yeah no problem.

**Pull request merged**